pintc/tests/arrays/bad_const_index.pnt

```diff
@@ -17,7 +17,7 @@ predicate test {
 // attempt to index an array with a mismatched value
 // @71..75: array access must be with an int value
 // found access using type `real`
-// constraint expression type error
-// @58..81: constraint expression has unknown type
-// @58..81: expecting type `bool`
+// binary operator type error
+// @80..81: operator `==` argument has unexpected type `int`
+// @69..76: expecting type `real`
```

Comment:
This shouldn't be an error right? `a[1.2]` is an `int` technically even though the index is wrong... Saying that we expect type `real` seems incorrect, no?